packages/aws-cdk-lib/aws-cloudfront-origins/README.md

```diff
@@ -1,26 +1,502 @@
 # CloudFront Origins for the CDK CloudFront Library
 
-
 This library contains convenience methods for defining origins for a CloudFront distribution. You can use this library to create origins from
 S3 buckets, Elastic Load Balancing v2 load balancers, or any other domain name.
 
 ## S3 Bucket
 
-An S3 bucket can be added as an origin. If the bucket is configured as a website endpoint, the distribution can use S3 redirects and S3 custom error
-documents.
+An S3 bucket can be used as an origin. An S3 bucket origin can either be configured as a standard bucket or as a website endpoint (see AWS docs for [Using an S3 Bucket](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/DownloadDistS3AndCustomOrigins.html#using-s3-as-origin)).
+
+### Standard S3 Bucket
+
+To set up an origin using a standard S3 bucket, use the `S3BucketOrigin` class. The bucket
+is handled as a bucket origin and
+CloudFront's redirect and error handling will be used. It is recommended to use `S3BucketOrigin.withOriginAccessControl()` to configure OAC for your origin.
+
+> Note: When you use CloudFront OAC with Amazon S3 bucket origins, you must set Amazon S3 Object Ownership to Bucket owner enforced (the default for new Amazon S3 buckets). If you require ACLs, use the Bucket owner preferred setting to maintain control over objects uploaded via CloudFront.
+
+```ts
+const myBucket = new s3.Bucket(this, 'myBucket');
+new cloudfront.Distribution(this, 'myDist', {
+  defaultBehavior: { origin: origins.S3BucketOrigin.withOriginAccessControl(myBucket) },
+});
+```
+
+### S3 Bucket Configured as a Website Endpoint
+
+To set up an origin using a S3 bucket configured as a website endpoint, use the `S3StaticWebsiteOrigin` class. When the bucket is configured as a
+website endpoint, the bucket is treated as an HTTP origin,
+and the distribution can use built-in S3 redirects and S3 custom error pages.
+
+```ts
+const myBucket = new s3.Bucket(this, 'myBucket');
+new cloudfront.Distribution(this, 'myDist', {
+  defaultBehavior: { origin: new origins.S3StaticWebsiteOrigin({ bucket: myBucket }) },
```

Comment:
From our discussion, there will be a `bucket` parameter added to the `S3StaticWebsiteOrigin` class. The reason it was taking in the object before was because it extends the `HttpOrigin` class which. accepts props as an object. This change will make the definitions more consistent.

Reply:
I added the bucket as a prop to `S3StaticWebsiteOriginProps` which extends `HttpOriginProps` , but I can update the API so the constructor it would look like:
```
origin: origins.S3BucketOrigin.withOriginAccessControl(myBucket)
origin: new origins.S3StaticWebsiteOrigin(myBucket)
```
which is more consistent